### PR TITLE
Feature/setup apple pay from buy products

### PR DESCRIPTION
**Pull request by prasanthu**
We are getting rid of membership from the product. So we need a way to make sure user has Apple pay. This patch will ask to setup Apple Pay (if necessary) at the time of buying a product.

### CODE REVIEW

File: ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+class SetupApplePayViewController: ApplePayViewController {
+
+    @IBOutlet private var buttonContainer: UIView!
+    @IBOutlet private var explanatoryCopyLabel: BodyTextLabel!
+
+    var paymentButton: PKPaymentButton?
+    var membership: Product?
+
+    lazy var linkableCopy: LinkableText = {
```
Comment (sgoodwin) [inline ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: Just for the record: there's no need to store this stuff anymore. The delegate method now informs which link was tapped, so we don't need to save it to match against.

File: ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        setupPaymentButton()
+    }
+
+    func setupPaymentButton() {
+        // Log information about the current Apple Pay setup.
+        let applePayError: ApplePayError? = canMakePayments()
+        switch applePayError {
+        case .unsupportedDevice?:
+            debugPrint("Apple Pay is not supported on this device")
```
Comment (sgoodwin) [inline ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: Do we need to do something during these errors?

File: ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        paymentButton.centerXAnchor.constraint(equalTo: buttonContainer.centerXAnchor).isActive = true
+        paymentButton.centerYAnchor.constraint(equalTo: buttonContainer.centerYAnchor).isActive = true
+        paymentButton.cornerRadius = 8.0
+        self.paymentButton = paymentButton
+    }
+
+    @objc func setUpButtonTapped() {
+        PKPassLibrary().openPaymentSetup()
+        // Go back to home screen.
+        dismiss(animated: true)
```
Comment (sgoodwin) [inline ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: Controllers generally should not be dismissing themselves. This assumes how the controller will be shown and can result in odd bugs later (especially when more than one controller is presented).

Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: Are you proposing the parent to dismiss this (in this case HomeViewController )? Also should this be done via a delegate telling the parent about this event ?

File: ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift (hunk 2 above)
Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: No, I will remove this. The object creating this should check for these things...

File: ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift (hunk 1 above)
Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: Can you suggest what I do here. I was just copying the structure in the "BecomeAMemberViewController"

Comment (sgoodwin) [reply ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: I mean you can create and assign the text directly. There's no need to make a property for this text.

File: ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift (hunk 3 above)
Comment (sgoodwin) [reply ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: Apple documentation recommends the controller that does the presenting handle the dismissing.

File: ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift (hunk 1 above)
Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: Remove `linkableCopy` property

File: ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift (hunk 3 above)
Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: It now uses protocols to convey the info back to the presenter

File: ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift (hunk 2 above)
Comment (prasanthu) [reply ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift]: Removed

**Pull request merged**